src/conduit-lwt-unix/conduit_lwt_unix_ssl.real.ml

```diff
@@ -52,22 +52,59 @@ module Client = struct
 
   let default_ctx = create_ctx ()
 
-  let connect ?(ctx = default_ctx) ?src ?hostname sa =
+  type verify = { hostname : bool; ip : bool }
+
+  let default_verify = { hostname = true; ip = false }
+
+  let verification { hostname; ip } = function
+    | None, _ when hostname -> invalid_arg "impossible to verify hostname"
+    | _, None when ip -> invalid_arg "impossible to verify ip"
+    | h, i ->
+        let hostname = if hostname && h <> None then h else None in
+        let ip = if ip && i <> None then i else None in
+        (hostname, ip)
+
+  let validate_hostname hostname =
+    try
+      let _ = Domain_name.(host_exn (of_string_exn hostname)) in
+      hostname
+    with Invalid_argument msg ->
+      let s =
+        Printf.sprintf "couldn't convert %s to a [`host] Domain_name.t: %s"
+          hostname msg
+      in
+      invalid_arg s
```

Comment:
This validate_hostname function does only ensure that the provided string is a `` [`host] Domain_name.t ``, this is its purpose?

Reply:
Yes, hostname can contain an IP